### ISSUE TITLE
Invalid parameter number for resource listings

### ISSUE THREAD

**Issue by andrewtweber** [laravel/nova-issues#3276]
- Laravel Version: 8.29.0
- Nova Version: 3.22.0
- PHP Version: 8.0.2
- Database Driver & Version: MySQL 5.7.25
- Operating System and Version: Ubuntu 16.04
- Browser type and version: Firefox 86.0
- Reproduction Repository: https://github.com/###/###

### Description:

Any of the index listings for any Nova resource fail with a PDO error. This seems to be an issue with Laravel's simple paginator

```
SQLSTATE[HY093]: Invalid parameter number (SQL: select * from clubs where clubs.deleted_at is null order by id limit 26 offset 0)
```

Stack trace:

```
Illuminate\\Database\\QueryException
vendor/laravel/framework/src/Illuminate/Database/Connection.php:678
vendor/laravel/framework/src/Illuminate/Database/Connection.php:638
vendor/laravel/framework/src/Illuminate/Database/Connection.php:346
vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2313
vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2301
vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2796
vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2302
vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:588
vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:572
vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:801
nova/src/Query/Builder.php:200
nova/src/Http/Requests/ResourceIndexRequest.php:30
nova/src/Http/Controllers/ResourceIndexController.php:20
... {routing trace} ...
```

Happens with or without soft deleting models

```
SQLSTATE[HY093]: Invalid parameter number (SQL: select * from amenities order by name limit 26 offset 0)
```

### Detailed steps to reproduce the issue on a fresh Nova installation:

* Install Nova
* Create a model and a related resource (e.g. "Club")
* Go to the resource's index listing (e.g. "/nova/resources/clubs")
* The API request to fetch the resources fails with a Server error

Individual resources can be viewed, but editing fails if it has a relationship and is trying to populate a dropdown with a list of resources

**Comment by crynobone** [comment]
Please create a full reproducing code since it's hard to follow the issue from just above. 

* `where clubs.deleted_at is null` will only be added if Model contains `SoftDeletes` scope.
* **Go to the resource's index listing (e.g. "/nova/resources/clubs")** and **editing fails if it has a relationship and is trying to populate a dropdown with a list of resources** are two different XHR request, and doesn't reproduce the same SQL queries as mention above.

**Comment by andrewtweber** [comment]
@crynobone figured this out, I was extending the base Resource class `indexQuery` with an `orderByRaw` where I should've been using `orderBy` so it was giving an incorrect number of parameters
